### PR TITLE
*: Txn() function signature refactor and remove ActivePendingTxn()

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Clean up after https://github.com/pingcap/tidb/pull/8260

### What is changed and how it works?

* change `Txn()` function signature to `Txn(active bool)`
* `ActivePendingTxn()` is not used any more because `Txn()` does the work
* change executor builder `getStartTS() uint64` to `getStartTS() (uint64, error)`

### Check List 

Tests 

 - No code

@zz-jason @jackysp @zimulala @winkyao

### CODE REVIEW

File: executor/prepared.go [hunk 1]
```diff
@@ -207,8 +207,6 @@ func (e *ExecuteExec) Build() error {
 	var err error
 	if IsPointGetWithPKOrUniqueKeyByAutoCommit(e.ctx, e.plan) {
 		err = e.ctx.InitTxnWithStartTS(math.MaxUint64)
-	} else {
-		err = e.ctx.ActivePendingTxn()
```
Comment (jackysp) [inline executor/prepared.go]: Is it missed in the last PR?

Comment (tiancaiamao) [reply executor/prepared.go]: I think, yes, Prepared Executor is forgot
https://github.com/pingcap/tidb/pull/8260/files#diff-350127760839dbfd52d23927f7ff2d95L290

**Comment by shenli** [comment]
I can not get the point of this PR. And for the Tests, why do you say "No code". There are a bunch of code changes.

**Comment by tiancaiamao** [comment]
`Txn(...bool)` this function signature is terrible, it was introduced as a compromise because I want the minimal changes. @shenli

**Pull request merged**